server/src/main/java/org/elasticsearch/index/mapper/MapperService.java

```diff
@@ -202,13 +202,20 @@ public void updateMapping(final IndexMetadata currentIndexMetadata, final IndexM
     }
 
     private boolean assertRefreshIsNotNeeded(DocumentMapper currentMapper,
-                                          String type,
-                                          CompressedXContent incomingMappingSource,
-                                          Mapping incomingMapping) {
+                                             String type,
+                                             Mapping incomingMapping) {
         Mapping mergedMapping = mergeMappings(currentMapper, incomingMapping, MergeReason.MAPPING_RECOVERY);
+
+        ToXContent.MapParams params = new ToXContent.MapParams(Collections.singletonMap("skip_runtime_section", "true"));
```

Comment:
Nice! Might be worth making `skip_runtime_section` a static final String on RuntimeFieldType and adding a comment there on why we need it?